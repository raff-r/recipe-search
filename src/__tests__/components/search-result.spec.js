import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";

import SearchResult from '../../components/search-result';
import mockResponse from '../mock-response.json';

describe('SearchResults Component', () => {

    let component;

    beforeEach(() => {
        component = shallow( <SearchResult result={mockResponse.hits[0].recipe} /> );
    });

    it('should match the snapshot', () => {
        expect(toJson(component)).toMatchSnapshot();
    });

});