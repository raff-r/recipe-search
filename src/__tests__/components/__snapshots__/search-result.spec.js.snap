// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults Component should match the snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-xs-12"
  >
    <h3>
      Herbes de Provence Rotisserie Chickens
    </h3>
  </div>
  <div
    className="col-xs-6 col-md-4"
  >
    <img
      alt="Herbes de Provence Rotisserie Chickens"
      className="img-responsive"
      src="https://www.edamam.com/web-img/18d/18dcf05995cb40e8ce4c077972341d7a.jpg"
    />
  </div>
  <div
    className="col-xs-6 col-md-3"
  >
    <h4>
      Diet labels
    </h4>
    <ul
      className="list-inline"
    >
      <li>
        <span
          className="label label-info"
        >
          Low-Carb
        </span>
      </li>
    </ul>
    <h4>
      Health labels
    </h4>
    <ul
      className="list-inline"
    >
      <li>
        <span
          className="label label-info"
        >
          Sugar-Conscious
        </span>
      </li>
      <li>
        <span
          className="label label-info"
        >
          Peanut-Free
        </span>
      </li>
      <li>
        <span
          className="label label-info"
        >
          Tree-Nut-Free
        </span>
      </li>
      <li>
        <span
          className="label label-info"
        >
          Alcohol-Free
        </span>
      </li>
    </ul>
  </div>
  <div
    className="col-xs-12 col-md-5"
  >
    <h4>
      Ingredients
    </h4>
    <ul>
      <li>
        2 tablespoons (1/4 stick) butter, room temperature
         - 
        28.40
        g
      </li>
      <li>
        2 tablespoons dried herbes de provence*
         - 
        5.40
        g
      </li>
      <li>
        1 tablespoon coarse kosher salt
         - 
        14.77
        g
      </li>
      <li>
        2 (3 1/2-pound) chickens
         - 
        3175.15
        g
      </li>
    </ul>
  </div>
</div>
`;
